src/lib/mongoBackend/mongoRegistrationCreate.cpp

```diff
@@ -114,16 +114,26 @@ static void setContextRegistrationVector(ngsiv2::Registration* regP, mongo::BSON
 
   for (unsigned int eIx = 0; eIx < regP->dataProvided.entities.size(); ++eIx)
   {
-    ngsiv2::EntID* eP = &regP->dataProvided.entities[eIx];
+    ngsiv2::EntID* eP                  = &regP->dataProvided.entities[eIx];
+    std::string    entityId            = (eP->id          != "")? eP->id : eP->idPattern;
+    std::string    entityIsPattern     = (eP->idPattern   != "")? "true" : "false";
+    std::string    entityType          = (eP->type        != "")? eP->type : eP->typePattern;
+    bool           entityTypeIsPattern = (eP->typePattern != "")? true : false;
 
-    if (eP->type == "")  // No type provided => all types
+    if (entityType == "")  // No entity type given - all types match
     {
-      entities.append(BSON(REG_ENTITY_ID << eP->id));
+      entityType = ".*";
+      entityTypeIsPattern = true;
     }
+
+    if ((entityIsPattern == "false") && (entityTypeIsPattern == false))
```

Comment:
Style

Reply:
Perhaps we should change the style guide ...
I just added 8 lines of '{' '}' and the whole thing got a whole lot less readable ... :(

Reply:
Changing style is out of this PR. Thus, let's adhere to it, pls.

Reply:
"Fixed" in 3dc4bcf